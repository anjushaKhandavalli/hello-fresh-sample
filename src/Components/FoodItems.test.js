import {fireEvent, getByTestId, render} from "@testing-library/react";
import FoodItems from "./Fooditems";

describe("FoodItems", () => {

    const mockItems = [
        {
        "id": "94298",
        "imageUrl": "https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg",
        "name": "3 Chicken Wings",
        "description": "Tender, Spicy and Juicy.",
        "recipeData": {
            "description": "There\u2019s nothing like the simple things in life - the smell of freshly cut grass, sitting outside on a nice sunny day, spending time with friends and family. Well here is a recipe that delivers simple culinary pleasures - some nice fresh fish with a crispy crust, crunchy potato wedges and some delightfully sweet sugar snap peas flavoured with cooling mint. Slip into something comfortable and relax into a delicious dinner!",
            "headline": "with Sweet Potato Wedges and Minted Snap Peas",
            "ingredients": [
                "375g Sweet Potatoes",
                "1 Tsp Paprika",
                "2 Tbsps Parmesan Cheese",
                "1 Lemon",
                "A Few Sprigs Thyme",
                "25g Panko Breadcrumbs",
                "1 Tbsp Butter",
                "2 Cod Fillets",
                "150g Sugar Snap Peas",
                "A Few Sprigs Mint",
                "75ml Sour Cream"
            ],
        }
    },
        {
            "id": "94301",
            "imageUrl": "https://cdn.pixabay.com/photo/2021/10/10/18/03/food-6697917_1280.jpg",
            "name": "Chicken Livers and Portuguese Roll",
            "description": "Chicken Livers Topped with PERi-PERi Sauce",
            "recipeData": {
                "description": "There\u2019s nothing like the simple things in life - the smell of freshly cut grass, sitting outside on a nice sunny day, spending time with friends and family. Well here is a recipe that delivers simple culinary pleasures - some nice fresh fish with a crispy crust, crunchy potato wedges and some delightfully sweet sugar snap peas flavoured with cooling mint. Slip into something comfortable and relax into a delicious dinner!",
                "headline": "with Sweet Potato Wedges and Minted Snap Peas",
                "ingredients": [
                    "375g Sweet Potatoes",
                    "1 Tsp Paprika",
                    "2 Tbsps Parmesan Cheese",
                    "1 Lemon",
                    "A Few Sprigs Thyme",
                    "25g Panko Breadcrumbs",
                    "1 Tbsp Butter",
                    "2 Cod Fillets",
                    "150g Sugar Snap Peas",
                    "A Few Sprigs Mint",
                    "75ml Sour Cream"
                ],
            }
        },]
    it("should render the items received",() => {
        const component = render(<FoodItems items={mockItems}/>)

        const { getAllByTestId } = component;

        expect(getAllByTestId("foodItemImage").length).toBe(2)
        expect(getAllByTestId("foodItemName").length).toBe(2)
        expect(getAllByTestId("foodItemDescription").length).toBe(2)
        expect(component).toMatchSnapshot();
    })

    it("should render recipe modal on click of foodItem", () => {
        const component = render(<FoodItems items={mockItems}/>)
        const { getByTestId } = component;

        fireEvent.click(getByTestId(mockItems[0].id))
        expect(getByTestId("recipeModal")).toBeInTheDocument();
    })

    it("should not render recipe modal when none of the foodItem selected", () => {
        const component = render(<FoodItems items={mockItems}/>)
        const { queryByTestId } = component;

        expect(queryByTestId("recipeModal")).not.toBeInTheDocument();
    })
})