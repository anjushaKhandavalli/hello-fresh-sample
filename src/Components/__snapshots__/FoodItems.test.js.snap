// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodItems should render the items received 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="foodItems"
        >
          <div
            class="foodItem"
            data-testid="94298"
          >
            <img
              alt="some food Image"
              data-testid="foodItemImage"
              height="200"
              src="https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg"
              width="360"
            />
            <p
              data-testid="foodItemName"
              id="foodItemName"
            >
              3 Chicken Wings
            </p>
            <p
              data-testid="foodItemDescription"
              id="foodItemDescription"
            >
              Tender, Spicy and Juicy.
            </p>
          </div>
          <div
            class="foodItem"
            data-testid="94301"
          >
            <img
              alt="some food Image"
              data-testid="foodItemImage"
              height="200"
              src="https://cdn.pixabay.com/photo/2021/10/10/18/03/food-6697917_1280.jpg"
              width="360"
            />
            <p
              data-testid="foodItemName"
              id="foodItemName"
            >
              Chicken Livers and Portuguese Roll
            </p>
            <p
              data-testid="foodItemDescription"
              id="foodItemDescription"
            >
              Chicken Livers Topped with PERi-PERi Sauce
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="foodItems"
      >
        <div
          class="foodItem"
          data-testid="94298"
        >
          <img
            alt="some food Image"
            data-testid="foodItemImage"
            height="200"
            src="https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg"
            width="360"
          />
          <p
            data-testid="foodItemName"
            id="foodItemName"
          >
            3 Chicken Wings
          </p>
          <p
            data-testid="foodItemDescription"
            id="foodItemDescription"
          >
            Tender, Spicy and Juicy.
          </p>
        </div>
        <div
          class="foodItem"
          data-testid="94301"
        >
          <img
            alt="some food Image"
            data-testid="foodItemImage"
            height="200"
            src="https://cdn.pixabay.com/photo/2021/10/10/18/03/food-6697917_1280.jpg"
            width="360"
          />
          <p
            data-testid="foodItemName"
            id="foodItemName"
          >
            Chicken Livers and Portuguese Roll
          </p>
          <p
            data-testid="foodItemDescription"
            id="foodItemDescription"
          >
            Chicken Livers Topped with PERi-PERi Sauce
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
